internal/otelcollector/config/gatewayprocs/k8s_attribute_proc.go

```diff
@@ -73,3 +85,26 @@ func extractLabels() []config.ExtractLabel {
 		},
 	}
 }
+
+func buildPodLabelPresets(presets EnrichmentOpts) []config.ExtractLabel {
+	podLabelPresets := make([]config.ExtractLabel, 0)
+
+	if presets.Enabled {
+		for _, label := range presets.PodLabels {
+			labelConfig := config.ExtractLabel{
+				From:    "pod",
+				TagName: "k8s.pod.label.$0",
+			}
+
+			if label.KeyPrefix != "" {
+				labelConfig.KeyRegex = fmt.Sprintf("(%s.*)", label.KeyPrefix)
+			} else {
+				labelConfig.KeyRegex = fmt.Sprintf("(^%s$)", label.Key)
+			}
+
+			podLabelPresets = append(podLabelPresets, labelConfig)
+		}
+	}
+
+	return podLabelPresets
```

Comment:
```suggestion
	return extractLabels
```
Let's not use the term presets here since you already renamed the API struct